### ISSUE TITLE
Similar to 'from_json(Column)' SparkSQL UDF, implement 'from_confluent_avro(Column)'

### ISSUE THREAD

**Issue by cricket007** [AbsaOSS/ABRiS#16]
From the earlier issues, I saw that both keys and values should be able to be deserialized as Avro (I think this was the initial design, or only the values are extracted? Didn't dig too deep there). 

Then there was discussion about keys are not Avro, and the values are Avro, which lead to funky classes like `ConfluentKafkaAvroWriterWithPlainKey`

As per discussion in #6 , I mentioned, what if I have a "plain value" and an avro key? What if my keys or values are some other datatype, not just strings and the other field _is Avro_?  

Rather than create one-off methods for each combination of supported Avro data-types, I feel like a better way to implement encoders/decoders would be to create individual UDF-type functions such as the [existing `from_json(Column, Map[String, String])` to `to_json(Column, Map[String, String])` functions in Spark](https://spark.apache.org/docs/2.2.0/api/scala/index.html#org.apache.spark.sql.functions$), where for Avro support, those option maps would include at least the schema registry url. 

From a usability perspective, I would expect this to work if I was doing "word count" on a topic with an Avro key. 

```scala
df.select(from_confluent_avro(col("key")), col("value").cast("int"))
```

**Comment by cricket007** [comment]
See https://issues.apache.org/jira/browse/SPARK-24811 for inspiration

**Comment by felipemmelo** [comment]
Tks a lot, Jordan, we'll start to look into this feature on Jan.

**Comment by cricket007** [comment]
Happy to see #48 😁

**Comment by cerveada** [comment]
This is implemented in version 3.0.0. It should be availabel at maven repository in few hours. 🙂

**Comment by cricket007** [comment]
Hi @cerveada , I think this might have been asked before, but any news on having a Scala 2.12 release?
